### DISCUSSION TITLE
AIS-catcher on Mac not getting serial AIS data

### DISCUSSION THREAD

**Discussion by Chrisapp22** [jvde-github/AIS-catcher#285] [Q&A] [answered]
Hi,

I'm trying to use AIS-catcher on my M3 Mac. I installed it via brew and it seems to run fine. To start it, I use:
`ais-catcher -e 115200 /dev/tty.usbserial-AB0PYFKO -N 8100`

The receiver is connected via USB-Serial, I verified the baudrate & port using a serial port viewer, where I see data just fine. AIS-catcher starts up fine as well, I can access the web interface.

However, I don't see any messages, it seems like AIS-catcher is getting no packets at all. The AIS receiver is indicating frames using an LED which is blinking frequently, so there should be plenty of data to see. I'm located directly at the north sea shore and can see a lot of ships right now ;)

I tried running it with sudo, but that did not help. The same receiver is working with AIS-catcher on Windows just fine.

Any idea what I could try to get it running?

Thanks
Chris

**Comment by jvde-github** [comment]
Hello,

Have you tried to run at 38400 or some other baudrate? You manage to connect this device to other programs on the Mac?

I will do some more checking for Mac specifically in the coming days otherwise.

Regards Jasper

**Reply by Chrisapp22** [reply]
Hi Jasper,

the receiver runs fine with a serial monitor at 115200, here's a screenshot that was taken on the same Mac:
![image](https://github.com/jvde-github/AIS-catcher/assets/31177778/c599cd6f-08ce-4009-92d9-0ecc68be546c)

So the data is there and seems to have the right format.

Let me know if you need more debug information.

BR

Chris

**Reply by jvde-github** [reply]
Hi Chris,

I will do some testing on my MacOS in the coming days.  It is older than an M3.

Perhaps the following workaround does the trick in the meantime?
```
stty -f /dev/tty.usbserial-AB0PYFKO 115200
cat /dev/tty.usbserial-AB0PYFKO | AIS-catcher -r txt . -N 8100
```
The cat is effectively similar to what AIS-catcher is doing, so I hope that is working.
One small thing, can you sheck the permissions?
```
ls -l /dev/tty.usbserial-AB0PYFKO
```

Thanks for raising this btw. Jasper

**Reply by Chrisapp22** [reply]
Hi Jasper,

using the provided stty command does not change anything.

Here are the permissions:
ls -l /dev/tty.usbserial-AB0PYFKO
crw-rw-rw-  1 root  wheel  0x9000002  8 Jul 12:30 /dev/tty.usbserial-AB0PYFKO

I confirmed this behavior on another M1 Mac, sadly I don't have any Intel machine to test on.

BR

Chris

**Reply by jvde-github** [reply]
Thank you. Will try to test on the Mac over the weekend with a Comar. 

Can you give me a hint on the USB-serial device? 

BR, Jasper

**Comment by jvde-github** [comment]
I managed to connect my Comar to a macos and get some output with `screen` but not with a straightforward `cat`. Will dive a bit more into this in the coming days.

**Comment by jvde-github** [comment]
Ok, I think the Mac code needs a few tweaks. Got it to work for Comar on MacOS but let you know once the production code has the necessary fixes.

**Reply by Chrisapp22** [reply]
Hi Jasper,

great, thanks for the update.

Let me know if I can support the testing in any way.

BR

Chris

**Reply by jvde-github** [reply, marked as answer]
Chris, uploaded a small patch to get, at least the Comar AIS receiver, working on MacOS: 
```
AIS-catcher -e 38400 /dev/tty.usbserial-FTQ9RAIS -ge print on
AIS-catcher (build Jul 15 2024) v0.59-216-g019c7e6b
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
>> !AIVDM,1,1,,B,13bVm>0P00PCQ>>Mksk>4?vl2H?b,0*54
!AIVDM,1,1,,B,13bVm>0P00PCQ>>Mksk>4?vl2H?b,0*54 ( MSG: 1, REPEAT: 0, MMSI: 246003000, timestamp: 20240715151627)
>> !AIVDM,1,1,,B,43aK`NivQog@MPBsifMo
>> BOA02@AU,0*3B
!AIVDM,1,1,,B,43aK`NivQog@MPBsifMoBOA02@AU,0*3B ( MSG: 4, REPEAT: 0, MMSI: 244770939, timestamp: 20240715151630)
```

Note that `-ge print on` is an optional argument for debugging that dumps the raw signal from the device to screen (after the >>). Appreciate if you can check if this resolves your issue as well.

**Reply by Chrisapp22** [reply]
Hi Jasper,

works great, thanks for the fast fix! I'm getting messages:
![image](https://github.com/user-attachments/assets/3318cbab-d73c-4678-9805-254994b94502)

Thanks again

Christopher
